lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -491,10 +503,94 @@ class TraceProcessor {
     // Filter to just events matching the frame ID, just to make sure.
     const frameEvents = keyEvents.filter(e => e.args.frame === mainFrameIds.frameId);
 
-    // Our navStart will be the last frame navigation in the trace
-    const navigationStart = frameEvents.filter(this._isNavigationStartOfInterest).pop();
-    if (!navigationStart) throw this.createNoNavstartError();
-    const timeOriginEvt = navigationStart;
+    // Compute our time origin to use for all relative timings.
+    const timeOriginEvt = this.computeTimeOrigin(
+      {keyEvents, frameEvents, mainFrameIds},
+      timeOriginDeterminationMethod
+    );
+
+    // Compute the key frame timings for the main frame.
+    const frameTimings = this.computeKeyTimingsForFrame(frameEvents, {timeOriginEvt});
+
+    // Subset all trace events to just our tab's process (incl threads other than main)
+    // stable-sort events to keep them correctly nested.
+    const processEvents = TraceProcessor
+      .filteredTraceSort(trace.traceEvents, e => e.pid === mainFrameIds.pid);
+
+    const mainThreadEvents = processEvents
+      .filter(e => e.tid === mainFrameIds.tid);
+
+    const frames = keyEvents
+      .filter(evt => evt.name === 'FrameCommittedInBrowser')
+      .map(evt => evt.args.data)
+      .filter(/** @return {data is {frame: string, url: string}} */ data => {
+        return Boolean(data && data.frame && data.url);
+      });
+
+    const traceOfTab = {...frameTimings, frames, mainThreadEvents, processEvents, mainFrameIds};
```

Comment:
it might be worth playing with code order permutations here? It's really nice having an object literal version of `traceOfTab` in code, and replacing it with a spread makes it so you can't see everything all in one place (and have to verify that there aren't overlaps in property names).

Updating the traceEnd timing info below (not because more info has been computed, but (I assume) due to awkwardness of `frameTimings` vs all the trace events) also suggests a more linear version is somewhere under the years of accumulated layers here :)

Reply:
Alright, made everything explicit 👍